app/lib/linters/spectral.rb

```diff
@@ -1,18 +1,24 @@
 module Linters
+  class SpectralLinterError < StandardError; end
   class Spectral
 
-    def initialize(file:)
+    def initialize(file:, system_command: Open3)
       @file = file
+      @system_command = system_command
     end
 
     def lint_to_json
-      # Use npx to run spectral-cli from the npm modules directory
-      # without it needed to be installed globally
-        `npx spectral lint -f json "#{file.path}"`
+
+      stdout_str, stderr_str = system_command.capture3("npx spectral lint -f json #{file.path}")
+      raise SpectralLinterError if stderr_str.present?
```

Comment:
I'd suggest passing the `stderr_str` to the error so that it gets logged:
```ruby
raise SpectralLinterError, stderr_str if stderr_str.present?
```

Reply:
This change has still not been implemented, but I think it would be worth including.